### PR TITLE
feat(destinations): Persistent queue support

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
### Proposed Change
* Update destinations with retry, queuing, and persistent queuing configuration
  * Prometheus remote write does not support queuing, WAL takes its place
  * Elasticsearch does not support the common retry options, it has it's own implementation of retry
  * Prometheus does not support either retry or queuing, due to its pull-based nature
  * Logzio destination doesn't support persistent queuing for metrics specifically (prom remote write), so WAL support was added for metrics.
* Modify existing render functions to take into account persistent queuing
* Remove indent parameter from existing render functions
* Fix signalfx destination having the "gce" key for resource detection instead of the "gcp" key (this was a change upstream in OTEL that we missed)

What configuration looks like:

Screen Shot 2023-04-25 at 12 45 23 PM

##### Checklist
- [ ] Changes are tested
- [ ] CI has passed

### CODE REVIEW

File: resources/destination-types/prometheus_remote_write.yaml [hunk 1]
```diff
@@ -231,3 +281,5 @@ spec:
             key_file: "{{ .key_file }}"
             {{ end }}
           {{ end }}
+          {{ bpRenderOtelRetryOnFailureConfig .retry_on_failure_enabled .retry_on_failure_initial_interval .retry_on_failure_max_interval .retry_on_failure_max_elapsed_time }} 
+
```
Comment (jsirianni) [inline resources/destination-types/prometheus_remote_write.yaml:285]: Might be a weird character on that last line.

File: resources/destination-types/logzio.yaml [hunk 2]
```diff
@@ -91,6 +91,182 @@ spec:
...
+      default: false
+      options:
+        sectionHeader: true
+      advancedConfig: true
+
+    - name: wal_storage_path
+      label: Storage Directory
+      description: Name to use for the directory the WAL is stored in. Must be unique to this destination.
+      type: string
+      default: logzio_metrics_wal
```
[the remark this answers is not in the document]
Comment (jsirianni) [reply resources/destination-types/logzio.yaml:108]: Looking at prom remote write, we do the same thing (just specifying a directory name).
```
directory: $OIQ_OTEL_COLLECTOR_HOME/storage/{{ .wal_storage_path }}
```

I still think we should use the full path for the value. We may want to update the remote write directory. Ill defer to your judgement here.

**Review by jsirianni — COMMENTED** [review]
Couple comments

I spoke with @ryancgoins and we think persistent queue should be enabled by default.

Also, we may be generating the retry_on_failure duration params wrong. We call out "seconds" but my config does not have seconds specified:
```
        retry_on_failure:
            enabled: true
            initial_interval: 5
            max_elapsed_time: 300
            max_interval: 30
```

This was probably already an issue.

I believe the underlying types are time.Duration, unless I am looking in the wrong place https://github.com/open-telemetry/opentelemetry-collector/blob/v0.76.0/exporter/exporterhelper/queued_retry.go#L249

File: resources/destination-types/logzio.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply resources/destination-types/logzio.yaml]: Yeah, I had copied the defaults and way it works from the prom remote write destination; I agree, it's confusing.

I think I'm gonna leave it as is in the prom remote write one, just so we don't break existing configurations there.

File: resources/destination-types/logzio.yaml [hunk 3]
```diff
@@ -91,6 +91,182 @@ spec:
...
+        sectionHeader: true
+      required: false
+      relevantIf:
+        - name: sending_queue_enabled
+          operator: equals
+          value: true
+    
+    - name: persistent_queue_directory
+      label: Persistent Queue Directory
+      description: The directory used to temporarily store telemetry data.
```
[the remark this answers is not in the document]
Comment (BinaryFissionGames) [reply resources/destination-types/logzio.yaml]: Clarified that it is "log and trace" data as opposed to just "telemetry" data.

**Review by jmwilliams89 — APPROVED** [review]

**Pull request merged**